app/main.py

```diff
@@ -1,3 +1,61 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int | float | Distance) -> None:
+        self.km = km
```

Comment:
Issue: The constructor does not handle the case where `km` is a `Distance` instance. According to the checklist, you should check if `km` is a `Distance` and extract its `.km` value, otherwise assign as is. This can lead to nested Distance objects or incorrect values.

Reply:
Так, дійно, в ініті Distance не потрібен
```suggestion
    def __init__(self, km: int | float) -> None:
        self.km = km
```